File: test/language/packages/cpp98/DefaultPackageTest.cpp
#include "gtest/gtest.h"

#include "DefaultPackageStructure.h"

TEST(DefaultPackageTest, defaultPackageStructure)
{
    // just test that DefaultPackageStructure is available in global namespace
    DefaultPackageStructure structure;
    structure.initialize(4);
    structure.setValue(10);
    default_package_import::top::TopStructure topStructure;
    topStructure.setType(1);
    topStructure.setData(1234);
    structure.setTopStructure(topStructure);
    Child child;
    child.setValue(0xdeadbeef);
    structure.setChildStructure(child);
    ASSERT_EQ(10, structure.getValue());
    ASSERT_EQ(1, structure.getTopStructure().getType());
    ASSERT_EQ(1234, structure.getTopStructure().getData());
    ASSERT_EQ(0xdeadbeef, structure.getChildStructure().getValue());
}

File: test/language/packages/python/DefaultPackageTest.py
import unittest

from testutils import getZserioApi

class DefaultPackageTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "default_package.zs", hasPackage=False)

    def testDefaultPackageStructure(self):
        # just test that DefaultPackageStructure is available in default package
        structure = self.api.DefaultPackageStructure(4)
        structure.setValue(10)
        structure.setTopStructure(self.api.default_package_import.top.TopStructure(1, 1234))
        structure.setChildStructure(self.api.Child(0xdeadbeef))
        self.assertEqual(10, structure.getValue())
        self.assertEqual(1, structure.getTopStructure().getType())
        self.assertEqual(1234, structure.getTopStructure().getData())
        self.assertEqual(0xdeadbeef, structure.getChildStructure().getValue())
